core/src/main/java/com/scalar/db/api/DistributedTransactionManagerMetadata.java

```diff
@@ -0,0 +1,142 @@
+package com.scalar.db.api;
+
+import com.google.common.base.MoreObjects;
+import com.scalar.db.exception.storage.ExecutionException;
+import edu.umd.cs.findbugs.annotations.SuppressFBWarnings;
+import java.util.Objects;
+import javax.annotation.Nullable;
+import javax.annotation.concurrent.Immutable;
+
+/** Metadata for a distributed transaction manager. */
+@Immutable
+public class DistributedTransactionManagerMetadata {
```

Comment:
`DistributedTransactionManagerMetadata` represents metadata for a distributed transaction manager. With it, you can also retrieve underlying distributed storage metadata by `getDistributedStorageMetadata()`.